### ISSUE TITLE
Entry options are greyed out is back in 4.2.3

### ISSUE THREAD

**Issue by adudek16** [keepassxreboot/keepassxc#3412]
It appears that #2838 #3240 is back with 4.2.3.

## Steps to Reproduce
1.    Searched for an entry
2.   copied either username or password using right click
3.    pasted into firefox (do not think app maters)
4.    switch to that app (to hit enter)
5.    switched back to keepassxc
6.    right click on item and only option that is not grayed out is "new entry"
7.   close keepassxc and reopen, everything it back

## Context
Have to close and reopen the app or open up about

## Debug Info
KeePassXC - Version 2.4.3
Revision: 5d6ef0c

Qt 5.12.3
Debugging mode is disabled.

Operating system: macOS Mojave (10.14)
CPU architecture: x86_64
Kernel: darwin 18.7.0

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- TouchID

Cryptographic libraries:
 libgcrypt 1.8.4

**Comment by droidmonkey** [comment]
Interesting.

**Comment by droidmonkey** [comment]
I cannot replicate this issue.

**Comment by adudek16** [comment]
I can . I will resupply the steps

**Comment by adudek16** [comment]
open keepassxc and database
search for value
right click, copy username 
right click on webpage and paste on website
right click on keepassxc to copy password and this is where I hit the issue.
I will note that when I click on keepassxc, it isn't in the front as expected for an active window and I wonder if this is the trigger.  
I am also able to replicate the issue by using the arrow keys in the webpage (ie. typing this in) to trigger the issue. I could do a video showing a test database

**Comment by adudek16** [comment]
actually, I can replicate it without using the keyboard or pasting anything. All I have to do is make the webpage window active, then go back to keypassxc and the issue is there.

**Comment by droidmonkey** [comment]
Can you do that video please, this is confounding me

**Comment by georgesnow** [comment]
i can reproduce this. it seems to do with the active window or key window. 
![kpxc-issue](https://user-images.githubusercontent.com/12751441/62572253-4924d580-b861-11e9-85ba-6b4930fe8023.gif)

**Comment by adudek16** [comment]
@georgesnow @droidmonkey  That is exactly what I saw in my follow up. My video would be the exact same as posted.

**Comment by droidmonkey** [comment]
That's so weird, why doesn't KeePassXC get focus when you right click? This is a macOS anomoly...

**Comment by adudek16** [comment]
I went back to 2.3.4 and I cannot replicate it there. In 2.4.0-beta1, I can.
Could it be related to 2583
Fix native hiding of macOS windows
<https://github.com/keepassxreboot/keepassxc/commit/f8c6c9f72ca4f280713befe6fdfe139b3b5a2b9a>
?

**Comment by georgesnow** [comment]
I believe the hidden and active state respond different based on the event. In this case a right mouse click. Doesn’t make the app active, which i think is normal behavior for macos

**Comment by adudek16** [comment]
This doesn't happen in 2.3.4, so something in 2.4.0-beta1 dealing with the
new menu setup or something. I can try this on an older version of macos
10.12 or 10.13

**Comment by georgesnow** [comment]
It seems like that context menu is expecting keepassxc to be the active app. Otherwise it’s disabled. Which would go to @adudek16 s point with the change in the most recent version

**Comment by droidmonkey** [comment]
Yes the context menu relies on focus of the app. This will likely not be fixed ever due to the fix breaking behavior that was fixed in 2.4.0. My recommendation is to properly focus the application and access the context menu.

**Comment by adudek16** [comment]
I have been trying to ensure that is the case but frequently keepass doesn't seem to want to be the active window even if I click on it first (left click), the right click to get the context menu, the items are still greyed out and I have to go back and forth a few time to get it to be active.

**Comment by georgesnow** [comment]
its subtle in the gif i posted, but i had to left click in a field (ie not the title bar) to make the window "active" and then right click to get the menu options to become available. It is possible to make right click contextual menu options available even if the window is not active, but in this case KPXC does not seem to active those menus. and it sounds like @droidmonkey referenced the fact it broke other things when those menus could be available when the app was not "active".

**Comment by adudek16** [comment]
I'll try your solution, but I definitely remember trying to click anywhere on the window without luck. I was using a dual monitor solution and I wonder if that contributed
 to the issue

**Comment by mjt772** [comment]
I'm experiencing this problem as well (on dual monitors). Left-clicking elsewhere in the KPXC window doesn't fix the problem for me; I can definitely focus the KPXC window (and even left-click on various other icons in the title bar), but the entry options remain grayed-out. Minimizing and re-expanding doesn't work. I've found only two ways to fix the issue: One, as others have mentioned,  is to close and re-launch the app. The other (and I believe this was also happening in the gif shared by @georgesnow) is to focus another window on top of (or at least overlapping) KPXC, and then click on KPXC again to refocus it; entry options then become available again. KPXC seems to get "stuck" that way, unable to be truly focused again until it's sharing the same screen real estate with another window and the OS needs to determine which one to put on top.

**Comment by adudek16** [comment]
now that you mention it i too have dual monitors when i see this issue

**Comment by droidmonkey** [comment]
This might be a Qt issue on mac

**Comment by georgesnow** [comment]
Interestingly enough I use dual monitor as well... I haven’t tested on my laptop or just single display, but I will now

**Comment by georgesnow** [comment]
tested on a single screen display and it still happens, but it was actually worse. In this case the menu even _after_ making the KPXC active and the focus window. The right context menu was still greyed out. I clicked (left and right) other entries, the toolbar, and search field (entered a search and selected an entry) and it was still greyed out. After selecting 2 other apps and some arbitrary amount of time (maybe 2-3 minutes??) when I left clicked on KPXC again to make it active and I right clicked an entry it worked again.

**Comment by yThomasS** [comment]
> tested on a single screen display and it still happens, but it was actually worse. In this case the menu even _after_ making the KPXC active and the focus window. The right context menu was still greyed out. I clicked (left and right) other entries, the toolbar, and search field (entered a search and selected an entry) and it was still greyed out. After selecting 2 other apps and some arbitrary amount of time (maybe 2-3 minutes??) when I left clicked on KPXC again to make it active and I right clicked an entry it worked again.

Same for me - and I am also using dual monitors on a Mac

**Comment by droidmonkey** [comment]
OK I can replicate this pretty easily, dual monitors is irrelevant to the issue.

If you have a GROUP selected (focus color) prior to switching to another application, then RIGHT CLICK select an ENTRY in the entry list, the application will NOT receive focus. Then if you attempt to focus the application while the context menu is STILL OPEN, the app gets into a state where focus cannot be gained (note the highlighted color of the entry). This is immediately fixed if you focus another application then focus KeePassXC again by LEFT CLICK anywhere.

Again, this enters a broken state when you try to focus the application while the context menu is open in an unfocused state. This is most definitely a Qt bug on macOS and I will report it to them.

**Comment by adudek16** [comment]
Has the issue been verified by the QT Team (or is there a QT bug id reference that can be added here?)

**Comment by droidmonkey** [comment]
I meant to make a bug report and didn't. I'll put that back on my list.

**Comment by droidmonkey** [comment]
I meant to make a bug report and didn't. I'll put that back on my list.

**Comment by droidmonkey** [comment]
Here is the Qt Bug Report: https://bugreports.qt.io/browse/QTBUG-82194

**Comment by tpfortree** [comment]
MacOS 11.6 iMac 2021 ，，，can copy user name ([cmd+B] is ok),bug cannot copy password ([cmd+C] is not working)
